### ISSUE TITLE
Codebase manager crashes when running the `pull` command to sync local with remote

### ISSUE THREAD

**Issue by rashadg1030** [unisonweb/unison#2044]
When remote is ahead of local and I try to `pull`, I get the following error:

```
.> pull git@github.com:rashadg1030/my-unison.git
 query "PRAGMA journal_mode=WAL;"                                     
(and crashed)

 query "PRAGMA journal_mode=WAL;" local codebase...
(and crashed)

unison: SQLite3 returned ErrorIO while attempting to perform prepare "PRAGMA journal_mode=WAL;": disk I/O error
```

Same error occurs on `trunk` and `topic/extended-sql-errors` branch.

---
Found this issue in #2014.

System Details:
```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
File type is `ext4`
`x86_64` architecture. I'm using an Acer laptop.

**Comment by aryairani** [comment]
@rashadg1030 Does it only happen when the remote is ahead of local?

**Comment by rashadg1030** [comment]
@aryairani Yes, it seems so.

**Comment by rashadg1030** [comment]
@aryairani Yes, I tested the `pull` command after doing a push to the repo. In each case, they errored out, so the state of the git repo hasn't changed at all since I ran these series of commands you suggested here: https://github.com/unisonweb/unison/pull/2014#issuecomment-849310735

EDIT: I misunderstood your comment. I'm gonna try and test `pull` when remote isn't ahead of local.

**Comment by rashadg1030** [comment]
I will check some of the code changes you made in `topic/extended-sql-errors` branch and go from there. I need to figure out where `ucm` uses `sqlite3` and see when and how the `pull` is executed.

**Comment by rashadg1030** [comment]
This is what happens when I try to pull when I don't have the code locally:
```
.> pull https://github.com/asoltysik/unison-md5:.releases._latest external.md5
unison: SQLite3 returned ErrorCan'tOpen while attempting to perform open "/home/rashad/.cache/unisonlanguage/gitfiles/https$x3A$$x2F$$x2F$github$dot$com$x2F$asoltysik$x2F$unison-md5/.unison/v2/unison.sqlite3": unable to open database file
```
I tried this on the `topic/extended-sql-errors` branch.

EDIT: Same thing happened on `trunk`. Not sure what the problem could be.

**Comment by aryairani** [comment]
Hi @rashadg1030, thanks for your help in diagnosing this.  Just to be clear, does `pull` ever work on your computer with these builds, in any situation?

I also sent you a message via Slack to try to set up a time to screen share and/or debug together, if you're open to it.

**Comment by rashadg1030** [comment]
@aryairani I believe I've used the `pull` command before and it worked fine, but I'm not 100% sure.

Okay great, I'll check Slack.

**Comment by aryairani** [comment]
Summarizing the Slack conversation: @rashadg1030 is going to try to create a minimal transcript, and then depending on what the transcript does on my system, we'll set up a call to debug tomorrow.

**Comment by aryairani** [comment]
> This is what happens when I try to pull when I don't have the code locally:
> 
> ```
> .> pull https://github.com/asoltysik/unison-md5:.releases._latest external.md5
> unison: SQLite3 returned ErrorCan'tOpen while attempting to perform open "/home/rashad/.cache/unisonlanguage/gitfiles/https$x3A$$x2F$$x2F$github$dot$com$x2F$asoltysik$x2F$unison-md5/.unison/v2/unison.sqlite3": unable to open database file
> ```
> 
> I tried this on the `topic/extended-sql-errors` branch.
> 
> EDIT: Same thing happened on `trunk`. Not sure what the problem could be.

I opened #2052 to track the `ErrorCan'tOpen` error and hid some of the comments related to it.
